File: src/test/java/org/neo/gomina/model/security/PasswordsTest.java
package org.neo.gomina.model.security;

import org.apache.commons.codec.binary.Base64;
import org.junit.Test;

import java.io.File;
import java.util.Scanner;

import static org.fest.assertions.Assertions.assertThat;

public class PasswordsTest {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Password:");
        String password = scanner.next();
        System.out.println(new String(Base64.encodeBase64(password.getBytes())));
    }

    @Test
    public void getRealPassword() throws Exception {
        Passwords passwords = new Passwords(new File("config/pass.properties.sample"));

        assertThat(new String(Base64.encodeBase64("super$$sec".getBytes()))).isEqualTo("c3VwZXIkJHNlYw==");
        assertThat(passwords.getRealPassword("@amokrane")).isEqualTo("mysecret");
        assertThat(passwords.getRealPassword("@vac")).isEqualTo("super$$sec");
    }

}

File: src/test/java/org/neo/gomina/model/security/PasswordsTest.kt
package org.neo.gomina.model.security

import org.apache.commons.codec.binary.Base64
import org.fest.assertions.Assertions.assertThat
import org.junit.Test
import java.io.File
import java.util.*

class PasswordsTest {

    @Test
    fun getRealPassword() {
        val passwords = Passwords(File("config/pass.properties"))

        assertThat(String(Base64.encodeBase64("super$\$sec".toByteArray()))).isEqualTo("c3VwZXIkJHNlYw==")
        assertThat(passwords.getRealPassword("@test")).isEqualTo("test")
        assertThat(passwords.getRealPassword("@majortom")).isEqualTo("super$\$sec")
    }

}

fun main(args: Array<String>) {
    val scanner = Scanner(System.`in`)
    print("Password:")
    val password = scanner.next()
    println(String(Base64.encodeBase64(password.toByteArray())))
}